### PR TITLE
Cast to strptime

### PR DESCRIPTION
**Pull request by toanngosy**
Malformed timestamp failed to catch at this [line](https://github.com/AMF-FLX/AMF-BASE-QAQC/blob/develop/processing/timestamp_checks.py#L195). We don't catch the error when we can't cast the last element of TIMESTAMP_START.
The solution uses the function `cast_as_datetime()` from the `ts_util` to cast the timestamp string.
Fixes Issue #23

## PR Self Evaluation

- [x] I have reviewed my code to check that it contains no sensitive information.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests or modified existing tests that prove my fix is effective or that my feature works
- [x] Existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by toanngosy** [comment]
I added the test for `check_timestamp_format` where we have a bad timestamp at various positions in `TIMESTAMP_START` and `TIMESTAMP_END`.
I added some bad files related to timestamp issues (index 75 to the end) into `file_check_expected_issues_ts3.csv` and changed `run_upload_check_manual_tests.py` so it works with the current test folder structure. 
All checks from `run_upload_check_manual_tests.py` are broken because in `DataReader.read_single_file()` we no longer have `fname= and checklog=`, so I make the change here to make it works as well.

**Review by dschristianson — CHANGES_REQUESTED** [review]
Hi @toanngosy. Thanks for this work, especially for updating the manual testing helper files!

Some requested changes inline. 

Also, a request to change the tests as follows:
- The unit tests are fine. I'd like to see the case we observed included in the tests. I.e., 000000 for DDHHMM. See the #23 for more details. 
- For the test file for the end-2-end tests, please include just one file that has the observed case. 

The idea is that we catch all the variants of malformed timestamp in the unit tests, and have a representative case for manual end-2-end. Otherwise checking every variant in the manual tests becomes unmanageable.

File: processing/timestamp_checks.py [hunk 1]
```diff
@@ -191,10 +190,9 @@ def check_forward_filled_timestamps(self, check_log):
         """
         qaqc_check = self.msg.get_display_check(check_log.getName())
         msg = self.msg.get_msg(check_log.getName(), 'CRITICAL')
-        last_ts_start = self.data['TIMESTAMP_START'][-1].decode('ascii')
-        last_ts_start = datetime.datetime.strptime(
-            last_ts_start, self.ts_util.PREFERRED_TS_FORMAT)
-        if last_ts_start > self.current_date:
+        last_ts_start = self.ts_util.cast_as_datetime(
+            self.data['TIMESTAMP_START'][-1])
+        if last_ts_start and last_ts_start > self.current_date:
```
[the remark this answers is not in the document]
Comment (toanngosy) [reply processing/timestamp_checks.py:195]: This is from function `check_forward_filled_timestamps()`, is called at the end of `driver()`, so at this point the data already has `ts_valid=True`. So I think we can safely assume that the timestamp here is always valid.

[the remark this answers is not in the document]
Comment (dschristianson) [reply processing/timestamp_checks.py:196]: Great. Then small suggestion to change new line 196 to
`if last_ts_start > self.current_date:`
As is, it implies that there is a case in which last_ts_start is None.

**Comment by dschristianson** [comment]
Thanks for addressing my comments, @toanngosy. Please rebase with the develop branch as I just merged in the other PR. Thanks!

**Comment by toanngosy** [comment]
> Thanks for addressing my comments, @toanngosy. Please rebase with the develop branch as I just merged in the other PR. Thanks!

I just rebased this PR

**Review by dschristianson — CHANGES_REQUESTED** [review]
Hi @toanngosy. Looking good! Please have a look at my last comment from Friday. New line 196 in the timestamp_check.py module.

**Comment by toanngosy** [comment]
Thanks, Danielle, I missed that comment. The new commit addressed that.

**Review by dschristianson — APPROVED** [review]
Thanks, @toanngosy!

**Pull request merged**